modules/apps/oauth2-provider/oauth2-provider-rest-spi/src/main/java/com/liferay/oauth2/provider/rest/spi/web/key/provider/WebKeyProviderAccessor.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.oauth2.provider.rest.spi.web.key.provider;
+
+/**
+ * @author Stian Sigvartsen
+ */
+public interface WebKeyProviderAccessor {
+
+	public WebKeyProvider getWebKeyProvider(long companyId, String clientId);
```

Comment:
did you mean (long companyId, String issuer)?

Reply:
Also this would imply, we support multiple issuers per company, which we don't.

Reply:
But we do, that's the thing. The `BearerTokenProvider.AccessToken` has a `setIssuer()` method.

Reply:
Supporting multiple issuers per company does not simply mean our customization has a setIssuer().
Issuer is not simply a string, but an id to a server. If one company supports multiple issuer, than what it means is each company is able to server multiple authorization servers.

I am not sure if that's what we want, or even if we want, we are ready with current implementation.

Again, I am very strong against the idea to consider customization option as part of core implementation, and it does not even matter if BearerTokenProvider.AccessToken has a setIssuer() or not at all...

Reply:
That's fine. It's a complicated subject around issuer, and I think I need to document the use cases driving my thinking so we can explore & validate separately. I will proposed a way forward in a separate comment on this PR.

Please can you share your thinking on what impact a `BearerTokenProvider.AccessToken.setIssuer()` call **should** have if we don't use it with the new SPI? So we can validate the consistency of the whole design.